0001-typedb/01-start/04-iam-schema.md

```diff
@@ -0,0 +1,302 @@
+---
+pageTitle: IAM schema explanation
+keywords: iam, typedb, typeql, tutorial, quickstart, application, app, example, sample, schema
+longTailKeywords: identity and access, identity and access management, iam schema explanation, typedb tutorial, typedb quickstart, learn typedb, sample app, sample application
+summary: Explanation of the IAM schema, widely used throughout TypeDB documentation.
+toc: false
+---
+
+# IAM schema explanation
+
+## Introduction
+
+**Identity and access management** (**IAM**) is a complex field that has seen massive growth over the past two decades 
+to meet user demand. IAM systems have become progressively more complicated to account for the variety of user 
+requirements and to ensure compatibility with a backlog of legacy systems, leading to the definition of standards by 
+governing bodies, such as [NIST](https://doi.org/10.6028/NIST.SP.800-63-3), in an attempt to unify and simplify the 
+state of IAM ecosystems, as well as to set out best practices. Modern IAM systems, such as the 
+[RSA’s Identity Governance and Administration Platform](https://www.rsa.com/solutions/identity-governance-and-administration/), 
+implement these models while providing APIs to existing IAM systems an organization might have in order to unify IAM 
+under a single management framework.
+
+<div class="note">
+[Note]
+The terminology used in the IAM [schema](03-quickstart.md#prepare-a-tql-file) is defined according to the sources 
+mentioned above and ISO/IEC 15408-1 standard within the context of an information system.
+</div>
+
+<div class="note">
+[Note]
+The Identity and Access Management database [schema](03-quickstart.md#prepare-a-tql-file) and accompanying 
+[dataset](03-quickstart.md#insert-data) described here are used in most examples throughout TypeDB documentation, 
+including the [Quickstart guide](03-quickstart.md).
+</div>
+
+## Overview
+
+As with any [TypeDB schema](../02-dev/02-schema.md), the IAM schema consists of:
+
+- Types
+  - Entity types
+  - Relation types
+  - Attribute types
+- Rules
+
+The [schema](../02-dev/02-schema.md) page will later provide more information to better understand the 
+terminology of TypeDB schemas.
+
+The main entity types in the IAM schema:
+
+- Subject — Active entity in the _system_ that performs operations on objects. 
+  - User — Human or IT entity possibly interacting with the _system_ from outside of the _system_ boundary.
+    - Person — User with `credentials`, `email`, and `full-name` attributes.
+  - User-group — A group of users that share the same role.
+- Object — Passive entity in the _system_, that contains or receives information, and upon which subjects perform
+  operations. 
+  - Resource — Anything usable or consumable in the _system_.
+    - File — File in some filesystem with `path` (to the file) and `size` (in KB) attributes.
+  - Resource-collection — A collection of resources on which the same operations can be performed.
```

Comment:
This is not correct. A resource collection is only a grouping of resources. It is fine to have multiple resources within a collection that have different valid operations. The insertion of `accesses` by the user define valid operations on a case-by-case basis (this can be somewhat automated), and these are irrespective of object hierarchy.

Reply:
That is done by analogy with user-groups. If user-groups combine users with the same role, then what is the factor of combining resources together? The most logical assumption is to combine objects by how they can be accessed. So if a user-group is being used to combine multiple users in a single permission assignment, then a resource-collection can be used to combine multiple resources in a single access assignment.

Reply:
Consider if you have a file hierarchy originating from a root directory. If you were to follow this rule, every object (both resource and collection) would have to have the same set of valid operations. This is trivially unfeasible as the `create file` operation is valid for a directory but not a file, and the root directory contains both. If you have multi-typed object (e.g. an object that is both a file and an invoice), then these would have to be segregated into different collections by this rule.

Reply:
The idea was not to set equal permissions but to be able to set permission for a whole group. Like `view` all objects that are in the root directory.
The fact that we can set access to a whole resource-collection (and all its members) doesn't mean we must do it always. The original description seems inaccurate now. Probably because the user-group description uses the vague term "role" instead of "the same permissions" (as per analogy).
I think I'll demote resource-collection to be just for grouping, but the general idea remains the same - the use case of this grouping is pretty much the same as with users IMHO. So we can set things like access, ownership, etc. to a group, instead of every member of a group. It looks lacking if we don't explain how we should use the groups.

```suggestion
  - Resource-collection — A group of resources.
```